src/main/java/xyz/nucleoid/creator_tools/workspace/editor/ParticleOutlineRenderer.java

```diff
@@ -19,14 +19,11 @@ public static void render(ServerPlayerEntity player, BlockPos min, BlockPos max,
         for (var edge : edges) {
             int length = edge.length();
 
-            int interval = 1;
-            if (length > 0) {
-                interval = MathHelper.clamp(length / Math.min(maxCount, length), 1, maxInterval);
-            }
+            int steps = Math.max(Math.min(length, maxCount), (length + maxInterval - 1) / maxInterval);
+            double interval = steps == 0 ? 0 : length / ((double) steps);
 
-            int steps = (length + interval - 1) / interval;
             for (int i = 0; i <= steps; i++) {
-                double m = (double) (i * interval) / length;
+                double m = i * interval / length;
```

Comment:
Nit: looks like that's now mostly `i / steps`, and `interval` shouldn't be needed anymore :)